fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
         return false;
     }
 
-    private Set<String> getPartitionsToRefreshForMaterializedView() {
+    private Set<String> getPartitionsToRefreshForMaterializedView(Map<String, String> properties)
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
+        Set<String> rangePartitionNames = null;
```

Comment:
rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
and check start and end in getPartitionNamesByRange

Reply:
check  in 
com.starrocks.sql.analyzer.MaterializedViewAnalyzer.MaterializedViewAnalyzerVisitor#visitRefreshMaterializedViewStatement